sorts/circle_sort.go

```diff
@@ -0,0 +1,47 @@
+package main
+ 
+import "fmt"
+ 
+func circle_sort(arr[] int, start int,end int, swaps int)int{
+	s := start
+	e := end
+	mid := (end - start)/2
```

Comment:
this is wrong, `(end-start)/2 != (start +end)/2`... I understand you are trying to avoid the overflow errors. The correct implementation would be `mid := start+(end-start)/2` or` mid := end-(end-start)/2`
```suggestion
	mid := start + (end-start)/2
```